runelite-client/src/main/java/net/runelite/client/plugins/poison/PoisonPlugin.java

```diff
@@ -117,64 +141,81 @@ protected void shutDown() throws Exception
 		poisonNaturalCure = null;
 		nextPoisonTick = null;
 		lastValue = -1;
+		lastDiseaseValue = -1;
 
+		clientThread.invoke(this::resetHealthIcon);
 	}
 
 	@Subscribe
 	public void onVarbitChanged(VarbitChanged event)
 	{
 		final int poisonValue = client.getVar(VarPlayer.POISON);
-		if (poisonValue == lastValue)
+		if (poisonValue != lastValue)
 		{
-			return;
-		}
-
-		lastValue = poisonValue;
-		nextPoisonTick = Instant.now().plus(Duration.of(POISON_TICK_MILLIS, ChronoUnit.MILLIS));
+			lastValue = poisonValue;
+			nextPoisonTick = Instant.now().plus(Duration.of(POISON_TICK_MILLIS, ChronoUnit.MILLIS));
 
-		final int damage = nextDamage(poisonValue);
-		this.lastDamage = damage;
+			final int damage = nextDamage(poisonValue);
+			this.lastDamage = damage;
 
-		envenomed = poisonValue >= VENOM_THRESHOLD;
+			envenomed = poisonValue >= VENOM_THRESHOLD;
 
-		if (poisonValue < VENOM_THRESHOLD)
-		{
-			poisonNaturalCure = Instant.now().plus(Duration.of(POISON_TICK_MILLIS * poisonValue, ChronoUnit.MILLIS));
-		}
-		else
-		{
-			poisonNaturalCure = null;
-		}
-
-		if (config.showInfoboxes())
-		{
-			if (infobox != null)
+			if (poisonValue < VENOM_THRESHOLD)
 			{
-				infoBoxManager.removeInfoBox(infobox);
-				infobox = null;
+				poisonNaturalCure = Instant.now().plus(Duration.of(POISON_TICK_MILLIS * poisonValue, ChronoUnit.MILLIS));
+			}
+			else
+			{
+				poisonNaturalCure = null;
 			}
 
-			if (damage > 0)
+			if (config.showInfoboxes())
 			{
-				final BufferedImage image = getSplat(envenomed ? SpriteID.HITSPLAT_DARK_GREEN_VENOM : SpriteID.HITSPLAT_GREEN_POISON, damage);
+				if (infobox != null)
+				{
+					infoBoxManager.removeInfoBox(infobox);
+					infobox = null;
+				}
 
-				if (image != null)
+				if (damage > 0)
 				{
-					infobox = new PoisonInfobox(image, this);
-					infoBoxManager.addInfoBox(infobox);
+					final BufferedImage image = getSplat(envenomed ? SpriteID.HITSPLAT_DARK_GREEN_VENOM : SpriteID.HITSPLAT_GREEN_POISON, damage);
+
+					if (image != null)
+					{
+						infobox = new PoisonInfobox(image, this);
+						infoBoxManager.addInfoBox(infobox);
+					}
 				}
 			}
 		}
+
+		final int diseaseValue = client.getVar(VarPlayer.DISEASE_VALUE);
+		if (diseaseValue != lastDiseaseValue && config.changeHealthIcon())
```

Comment:
move the config check to checkHealthIcon, and update lastDiseaseValue even if the config is disabled